reference/promotions.v3.yml

```
@@ -195,6 +195,72 @@ paths:
         '204':
           description: The deletion was successful or the specified resource does not exist.
           content: {}
+
+  '/promotions/{promotion_id}/codegen':
+    post:
+      tags:
+        - Coupon Codes (Bulk)
+      summary: Generate Bulk Coupon Codes
+      description: |-
+        Generate a batch of coupon codes for a particular bulk coupon promotion.
+        
+        **Note:**
+        * batch_size is limited to 250 codes per request for first version. If batch_size is not an integer or larger than 250, it will return a 422 error code.
+        * The default rate limit for this endpoint is 40 concurrent requests.
+      operationId: generatePromotionCodesBatch
+      parameters:
+        - $ref: '#/components/parameters/ContentType'
+        - $ref: '#/components/parameters/Accept'
+        - $ref: '#/components/parameters/PromotionIdPath'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              required:
+                - batch_size
+              type: object
+              properties:
+                batch_size:
+                  type: integer
+                  description: The number of coupon codes to generate in each batch. The maximum value is 250.
+                  example: 5
+                  minimum: 1
+                  maximum: 250
+                max_uses:
+                  type: integer
+                  description: The maximum number of times each coupon code can be used. The default value is 1.
+                  example: 10
+                  minimum: 0
+                  maximum: 100000
+                max_uses_per_customer:
+                  type: integer
+                  description: The maximum number of times a specific customer can use each coupon code. The default value is 1.
+                  example: 5
+                  minimum: 0
+                  maximum: 100000
+        required: true
+      responses:
+        '201':
+          $ref: '#/components/responses/BulkCouponCodesResponse'
+        '400':
+          description: Invalid request.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse400'
+        '403':
+          description: The request payload was invalid.
```

seems like we need to update the 403 response of other endpoints in this file as well

```suggestion
          description: Forbidden
```